### PR TITLE
Feature/gssapi

### PR DESCRIPTION
**Pull request by dothebart**
### Scope & Purpose
This PR adds SPNEGO / GSSAPI support for the clients, and adds SPNEGO support into the server. 
The server part itself is going to be part of the enterprise offering and is still WIP.

- [x] Strictly *new functionality* (i.e. a new feature / new option, no need for porting)
- [x] The behavior in this PR can be (and was) *manually tested* (support / qa / customers can test it)
- [ ] Automatic tests will require special additions to our test environment, still to be done.

### Testing & Verification
These are going to be tested in a similar way to the LDAP authentification, and thus require additional Oskar support

### Documentation
TBD
- [ ] Added a *Changelog Entry* (referencing the corresponding public or internal issue number)
- [ ] Added entry to *Release Notes* 
- [ ] Added a new section in the *Manual* 
- [ ] Added a new section in the *http API*

### CODE REVIEW

File: arangosh/Shell/V8ClientConnection.cpp [hunk 1]
```diff
@@ -66,25 +72,222 @@ V8ClientConnection::V8ClientConnection()
       _vpackOptions(VPackOptions::Defaults) {
   _vpackOptions.buildUnindexedObjects = true;
   _vpackOptions.buildUnindexedArrays = true;
+  /*
   _builder.onFailure([this](fuerte::Error error, std::string const& msg) {
```
Comment (dothebart) [inline arangosh/Shell/V8ClientConnection.cpp]: This pruned the error message set in the application code by an empty message. Imho it doesn't surve any purpose and should kompletely be removed?

File: 3rdParty/fuerte/include/fuerte/message.h [hunk 2]
```diff
@@ -226,6 +226,8 @@ class Response final : public Message {
   Response(ResponseHeader&& reqHeader = ResponseHeader())
       : header(std::move(reqHeader)), _payloadOffset(0) {}
 
+  Response(Response const& resp);
```
Comment (graetzer) [inline 3rdParty/fuerte/include/fuerte/message.h]: no, just no

File: 3rdParty/fuerte/include/fuerte/message.h [hunk 3]
```diff
@@ -270,7 +272,14 @@ class Response final : public Message {
...
                   std::size_t payloadOffset);
 
+  void addSupportedAuth(std::string const& value);
+  bool isMethodAllowed(AuthenticationType meth) {
+    return std::find(_supportedAuths.begin(),
+                     _supportedAuths.end(),
+                     meth) != _supportedAuths.end();
+  }
  private:
+  std::vector<AuthenticationType> _supportedAuths;
```
Comment (graetzer) [inline 3rdParty/fuerte/include/fuerte/message.h]: no wtf

Comment (graetzer) [reply 3rdParty/fuerte/include/fuerte/message.h]: just use an enum if it has to be done

Comment (dothebart) [reply 3rdParty/fuerte/include/fuerte/message.h]: AuthenticationType is an enum? Rather a bitfield?

File: 3rdParty/fuerte/include/fuerte/message.h (hunk 2 above)
Comment (dothebart) [reply 3rdParty/fuerte/include/fuerte/message.h]: ok. How to get the reply of an error so proper error messages can be emitted or analyzed?

**Comment by KVS85** [comment]
Closed as outdated.

**Pull request closed without merging**